pyproject.toml

```diff
@@ -39,6 +39,7 @@ dependencies = [
     "ruamel.yaml",
     "scmrepo",
     "psutil",
+    "py3nvml"
```

Comment:
It looks like this is quite old an unmaintained. Any reason not to use `pynvml` from https://pypi.org/project/nvidia-ml-py/? Also, we need to decide if we always install this or make it optional.